src/python/bot/testcase_manager.py

```diff
@@ -512,21 +512,25 @@ def __init__(self,
     self._gestures = gestures
     self._needs_http = needs_http
 
-    engine_impl = engine.get(fuzzer_name)
-    if engine_impl:
+    fuzz_target = data_handler.get_fuzz_target(fuzzer_name)
+    if fuzz_target:
+      engine_impl = engine.get(fuzz_target.engine)
+      if not engine_impl:
+        raise RuntimeError(engine_impl)
```

Comment:
Why engine_impl in argument as it is None ? Maybe a message is better 'Engine not found or something'

Reply:
good catch

Reply:
Oops, fixed.